app/main/routes.py

```diff
@@ -16,20 +16,33 @@ def index():
     """Implement the index route for the main blueprint - serves as the homepage."""
     generate_slang_form = GenerateSlangForm()
     meaning_form = MeaningForm()
+    model_form = ChooseModelField()
 
     # Frequently used keyword arguments for rendering templates.
     kwargs = {
         "title": "Home",
         "generate_slang_form": generate_slang_form,
         "meaning_form": meaning_form,
+        "model_form": model_form,
     }
 
+    if model_form.validate():
+        print(f"Was selected : {model_form.select_model.data}")
+        model_type = model_form.select_model.data
+        model_form.select_model.object_data = model_type
+        session["model_type"] = model_type
+
+    # Persist model_type through re-clicking generate_slang and other refreshing
+    if "model_type" in session:
+        model_form.select_model.object_data = session["model_type"]
+        model_form.select_model.data = session["model_type"]
```

Comment:
Is this to keep the selection the same in the model_form such that it doesn't accidentally overwrite to another model?

Reply:
I don't know, without this, after pressing 'generate word' the model_type is reset to the default ("straattaal") regardless of what it was. I think it's because the forms get reinitialized, maybe there is a prettier way though.